### PR TITLE
Fixed Cached Functional Tests

### LINKED ISSUE #1239: Cached functional tests failing
# 🐛 Summary #

Functional tests using RunCached are failing due to naming change of the Invoke-RunCached cmdlet being changed in an update to Invoke-ScubaRunCached.  Need to update functional test orchestration to use the new cmdlet name.

## To reproduce ##

1. Review functional test failures

## Expected behavior ##

Functional tests that use RunCached pass.

### PR DESCRIPTION
**Pull request by julianjburgos**
#  #
  
## 🗣 Description ##
Fixed the Functional Testing test plans that originally had the original naming convention, RunCached, to be aligned with the new non-generic cmdlet name, ScubaCached

## 💭 Motivation and context ##
Cached Functional Tests were failing, with this fix, they should pass.

Closes #1239

## 🧪 Testing ##
Run the Cached Functional Tests

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] PR targets the correct parent branch (e.g., main or release-name) for merge.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] Changes are sized such that they do not touch excessive number of files.
- [x] *All* future TODOs are captured in issues, which are referenced in code comments.
- [x] These code changes follow the ScubaGear [content style guide](https://github.com/cisagov/ScubaGear/blob/main/CONTENTSTYLEGUIDE.md).
- [x] Related issues these changes resolve are linked preferably via [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [x] All relevant type-of-change labels added.
- [x] All relevant project fields are set.
- [x] All relevant repo and/or project documentation updated to reflect these changes.
- [x] Unit tests added/updated to cover PowerShell and Rego changes.
- [x] Functional tests added/updated to cover PowerShell and Rego changes.
- [ ] All relevant functional tests passed.
- [x] All automated checks (e.g., linting, static analysis, unit/smoke tests) passed.

## ✅ Pre-merge checklist ##

- [ ] PR passed smoke test check.
- [ ] Feature branch has been rebased against changes from parent branch, as needed

  Use `Rebase branch` button below or use [this](https://www.digitalocean.com/community/tutorials/how-to-rebase-and-update-a-pull-request) reference to rebase from the command line.
- [ ] Resolved all merge conflicts on branch
- [ ] Notified merge coordinator that PR is ready for merge via comment mention

## ✅ Post-merge checklist ##

- [ ] Feature branch deleted after merge to clean up repository.
- [ ] Verified that all checks pass on parent branch (e.g., main or release-name) after merge.

### CODE REVIEW

File: Testing/Functional/Products/TestPlans/aad.g5.testplan.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ProductName: aad
 TestPlan:
   - PolicyId: MS.AAD.2.1v1
-    TestDriver: RunCached
+    TestDriver: ScubaCached
```
Comment (gdasher) [inline Testing/Functional/Products/TestPlans/aad.g5.testplan.yaml:4]: Please take a look here: https://github.com/cisagov/ScubaGear/blob/ab4036e4801cf9e03c4d5c786bf70264e2701a27/Testing/Functional/Products/Products.Tests.ps1#L252

I think you will need to match the case.

also, please add a comment to that file as James suggested clarifying that those strings need to match "TestDriver" in the test plans.

**Comment by tkol2022** [comment]
For this PR to be approved, I would like to see screenshot evidence of successful execution of all the affected test plans against all tenant types. Keep in mind that for AAD you will be executing two types per tenant. One vanilla and one variant (either G3 or G5).

File: Testing/Functional/Products/Products.Tests.ps1 [hunk 2]
```diff
@@ -249,8 +249,9 @@ Describe "Policy Checks for <ProductName>"{
                 SetConditions -Conditions $Preconditions.ToArray()
                 RunScuba
             }
-            elseif ('SCuBACached' -eq $TestDriver){
-                Write-Debug "Driver: SCuBACached"
+            # Cmdlet names should match Test Plan drivers e.g. Driver should be ScubaCached instead of RunCached to be aligned with the refactor of Scuba specific Cmdlets
```
Comment (gdasher) [inline Testing/Functional/Products/Products.Tests.ps1:252]: (nit) "the refactor" is a point in time comment that will not make sense to someone a few years down the road (which refactor? we did a refactor? etc).  Instead, I would add a comment at the top of this if statement (e.g. line 222) that says something like:

\# Select which TestDriver to use for a given test plan.  TestDriver names (e.g. RunScuba, ScubaCached) must
\# match exactly (including case) the ones used in TestPlans.

**Review by gdasher — APPROVED** [review]
One small remaining change.

Tests look to be passing more like they were before: https://github.com/cisagov/ScubaGear/actions/runs/10150058265/job/28067189934

**Comment by dagarwal-mitre** [comment]
@nanda-katikaneni this should be ready to be merged

**Pull request merged**